### PR TITLE
`azurerm_netapp_*` - support for `tags` & add ID parsing functions

### PR DESCRIPTION
**Pull request by aqche**
Fixes: #5532 

Adds the `tags` option for NetApp resources.

I wrote the code, but I didn't realize you need to request access to NetApp before you can create resources, so I an unable to run the tests atm. I'll put in a request, but if someone has time to run the tests that would be helpful as well!

### CODE REVIEW

File: azurerm/internal/services/netapp/resource_arm_netapp_snapshot.go [hunk 1]
```diff
@@ -153,7 +158,45 @@ func resourceArmNetAppSnapshotRead(d *schema.ResourceData, meta interface{}) err
...
-	return nil
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceArmNetAppSnapshotUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).NetApp.SnapshotClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
```
Comment (aqche) [inline azurerm/internal/services/netapp/resource_arm_netapp_snapshot.go]: i realize the team is creating parsers for resource ids, would you like me to go ahead and create parsers for each of the netapp resoures or let that be addressed in another PR?

File: azurerm/internal/services/netapp/tests/resource_arm_netapp_volume_test.go [hunk 2]
```diff
@@ -437,6 +447,10 @@ resource "azurerm_netapp_volume" "test" {
     unix_read_only    = false
     unix_read_write   = true
   }
+
+  tags = {
+    "foo" = "bar"
```
Comment (katbyte) [inline azurerm/internal/services/netapp/tests/resource_arm_netapp_volume_test.go]: Could we ensure capitals work
```suggestion
    "FoO" = "BaR"
```

File: azurerm/internal/services/netapp/tests/resource_arm_netapp_volume_test.go [hunk 3]
```diff
@@ -366,6 +372,10 @@ resource "azurerm_netapp_volume" "test" {
     unix_read_only  = true
     unix_read_write = false
   }
+
+  tags = {
+    "foo" = "bar"
```
Comment (katbyte) [inline azurerm/internal/services/netapp/tests/resource_arm_netapp_volume_test.go]: Could we ensure capitals work
```suggestion
    "FoO" = "BaR"
```

File: azurerm/internal/services/netapp/tests/resource_arm_netapp_snapshot_test.go [hunk 4]
```diff
@@ -204,6 +219,30 @@ resource "azurerm_netapp_snapshot" "test" {
...
+resource "azurerm_netapp_snapshot" "test" {
+  name                = "acctest-NetAppSnapshot-%d"
+  account_name        = azurerm_netapp_account.test.name
+  pool_name           = azurerm_netapp_pool.test.name
+  volume_name         = azurerm_netapp_volume.test.name
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+
+  tags = {
+    "foo" = "baz"
```
Comment (katbyte) [inline azurerm/internal/services/netapp/tests/resource_arm_netapp_snapshot_test.go]: Could we ensure capitals work
```suggestion
    "FoO" = "BaR"
```

File: azurerm/internal/services/netapp/tests/resource_arm_netapp_snapshot_test.go [hunk 5]
```diff
@@ -204,6 +219,30 @@ resource "azurerm_netapp_snapshot" "test" {
   volume_name         = azurerm_netapp_volume.test.name
   location            = azurerm_resource_group.test.location
   resource_group_name = azurerm_resource_group.test.name
+
+  tags = {
+    "foo" = "bar"
```
Comment (katbyte) [inline azurerm/internal/services/netapp/tests/resource_arm_netapp_snapshot_test.go]: Could we ensure capitals work
```suggestion
    "FoO" = "BaR"
```

File: azurerm/internal/services/netapp/tests/resource_arm_netapp_pool_test.go [hunk 6]
```diff
@@ -225,6 +230,10 @@ resource "azurerm_netapp_pool" "test" {
   resource_group_name = azurerm_resource_group.test.name
   service_level       = "Standard"
   size_in_tb          = 15
+
+  tags = {
+    "foo" = "bar"
```
Comment (katbyte) [inline azurerm/internal/services/netapp/tests/resource_arm_netapp_pool_test.go]: Could we ensure capitals work
```suggestion
    "FoO" = "BaR"
```

File: azurerm/internal/services/netapp/tests/resource_arm_netapp_account_test.go [hunk 7]
```diff
@@ -234,6 +239,10 @@ resource "azurerm_netapp_account" "test" {
     domain              = "westcentralus.com"
     organizational_unit = "OU=FirstLevel"
   }
+
+  tags = {
+    "foo" = "bar"
```
Comment (katbyte) [inline azurerm/internal/services/netapp/tests/resource_arm_netapp_account_test.go]: Could we ensure capitals work
```suggestion
    "FoO" = "BaR"
```

File: azurerm/internal/services/netapp/resource_arm_netapp_volume.go [hunk 8]
```diff
@@ -226,6 +230,7 @@ func resourceArmNetAppVolumeCreateUpdate(d *schema.ResourceData, meta interface{
 			UsageThreshold: utils.Int64(storageQuotaInGB),
 			ExportPolicy:   expandArmNetAppVolumeExportPolicyRule(exportPolicyRule),
 		},
+		Tags: tags.Expand(t),
```
Comment (katbyte) [inline azurerm/internal/services/netapp/resource_arm_netapp_volume.go]: _minor_ no need to change it but this is preferred
```suggestion
		Tags: tags.Expand(d.Get("tags").(map[string]interface{})),
```

File: azurerm/internal/services/netapp/resource_arm_netapp_snapshot.go [hunk 9]
```diff
@@ -153,7 +158,45 @@ func resourceArmNetAppSnapshotRead(d *schema.ResourceData, meta interface{}) err
...
+	name := id.Path["snapshots"]
+
+	t := d.Get("tags").(map[string]interface{})
+
+	parameters := netapp.SnapshotPatch{
+		Tags: tags.Expand(t),
+	}
+
+	_, err = client.Update(ctx, parameters, resourceGroup, accountName, poolName, volumeName, name)
+	if err != nil {
```
Comment (katbyte) [inline azurerm/internal/services/netapp/resource_arm_netapp_snapshot.go]: these two lines can be merged

File: azurerm/internal/services/netapp/resource_arm_netapp_snapshot.go [hunk 10]
```diff
@@ -153,7 +158,45 @@ func resourceArmNetAppSnapshotRead(d *schema.ResourceData, meta interface{}) err
...
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, accountName, poolName, volumeName, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving NetApp Snapshot %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if resp.ID == nil || *resp.ID == "" {
+		return fmt.Errorf("Cannot read NetApp Snapshot %q (Resource Group %q) ID", name, resourceGroup)
+	}
+	d.SetId(*resp.ID)
```
Comment (katbyte) [inline azurerm/internal/services/netapp/resource_arm_netapp_snapshot.go]: we should only be setting ID on create

File: azurerm/internal/services/netapp/resource_arm_netapp_snapshot.go (hunk 1 above)
Comment (katbyte) [reply azurerm/internal/services/netapp/resource_arm_netapp_snapshot.go]: It would be preferred to do it now but either or is fine 🙂

File: azurerm/internal/services/netapp/resource_arm_netapp_snapshot.go [hunk 11]
```diff
@@ -94,9 +97,11 @@ func resourceArmNetAppSnapshotCreate(d *schema.ResourceData, meta interface{}) e
 	}
 
 	location := azure.NormalizeLocation(d.Get("location").(string))
+	t := d.Get("tags").(map[string]interface{})
 
 	parameters := netapp.Snapshot{
 		Location: utils.String(location),
+		Tags:     tags.Expand(t),
```
Comment (katbyte) [inline azurerm/internal/services/netapp/resource_arm_netapp_snapshot.go]: _minor_ no need to change it but this is preferred
```suggestion
		Tags: tags.Expand(d.Get("tags").(map[string]interface{})),
```

File: azurerm/internal/services/netapp/resource_arm_netapp_snapshot.go (hunk 1 above)
Comment (aqche) [reply azurerm/internal/services/netapp/resource_arm_netapp_snapshot.go]: sounds good, i'll get parsers added in

**Comment by aqche** [comment]
@katbyte thanks for the review! made the updates you suggested and added the parse functions. I still don't have access to NetApp but here are the parse function test results:

```
go test .
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/netapp/parse        0.006s
```
